### ISSUE TITLE
Bug: Cannot find module '@fortawesome-internal-tools/fontawesome-icons/canonical'

### ISSUE THREAD

**Issue by dg-robinson** [FortAwesome/Font-Awesome#20307]
### Bug description

I've just updated from 6.5.2 to 6.6.0 and I've started getting this error in every file where I import icons, which I'm doing like this:

```
import { light } from "@fortawesome/fontawesome-svg-core/import.macro";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
```

I can update all of the style packages to 6.6.0, but if I update `@fortawesome/fontawesome-svg-core` then I start getting this issue. The stack trace is pointing to the imports in `node_modules/@fortawesome/fontawesome-svg-core/import.macro.js`.

This is happening in a React app with Craco and Babel, and I'm importing the packages via npm.

### Font Awesome version

v6.6.0

### Serving

Other (as specified in the bug description)

### Implementation

SVG+JS

### Browser and Operating System

Chrome 126.0.6478.183 on macOS 14.5

### Web bug report checklist

- [ ] I have included a test case because my odds go _way_ up that the team can fix this when I do
- [X] I have [searched for existing issues](https://github.com/FortAwesome/Font-Awesome/issues) and to the best of my knowledge this is not a duplicate

**Comment by tagliala** [comment]
Hi!

Thanks for being part of the Font Awesome Community and thanks for reporting this.

Looks like a legit issue, let's assign @robmadole 

Ref:
https://github.com/FortAwesome/Font-Awesome/blob/37eff7fa00de26db41183a3ad8ed0e9119fbc44b/js-packages/%40fortawesome/fontawesome-svg-core/import.macro.js#L3-L4

**Comment by robmadole** [comment]
Yep, sure enough that's an internal tool that leaked into this module. We'll get this looked at and fixed as soon as we can.

**Comment by allsyntax** [comment]
Is this resolved? How do fix it? I'm seeing the same issue.

**Comment by tagliala** [comment]
Not fixed yet, sorry. Please keep an eye on this issue and wait for it to be closed

**Comment by CarrieWDing** [comment]
Is there a list of @fortawesome/fontawesome-svg-core versions impacted by the current error?

**Comment by tagliala** [comment]
99% confident that it's just 6.6.0

**Comment by dg-robinson** [comment]
> Is there a list of @fortawesome/fontawesome-svg-core versions impacted by the current error? 

Yep, only 6.6.0. It worked fine in 6.5.2 and below. You can still upgrade the individual icon libraries to 6.6.0 though.

**Comment by mbrown1101** [comment]
Hey guys, is there an update on this or a workaround? We have 6.5.2 installed and are having this issue.

**Comment by CarrieWDing** [comment]
We resolved the issue by removing the `^` symbol before the version number in the `package.json`, and then reinstalling the `node_modules`.  Hope it helps.

**Comment by CarrieWDing** [comment]
> Hey guys, is there an update on this or a workaround? We have 6.5.2 installed and are having this issue.

We resolved the issue by removing the `^` symbol before the version number in the `package.json`, and then reinstalling the `node_modules`.  Hope it helps.

**Comment by tagliala** [comment]
Hi,

as per https://github.com/FortAwesome/Font-Awesome/issues/20324#issuecomment-2277919071, can you double check your pageview usage?

**Comment by jeffgaynor** [comment]
@CarrieWDing is correct, the version needs to be locked to `6.5.2` for @fortawesome/fontawesome-svg-core only, so remove the `^`. Here's a full example of the workaround: 
```json
   "@fortawesome/fontawesome-svg-core": "6.5.2",
    "@fortawesome/free-brands-svg-icons": "^6.6.0",
    "@fortawesome/pro-duotone-svg-icons": "^6.6.0",
    "@fortawesome/pro-light-svg-icons": "^6.6.0",
    "@fortawesome/pro-regular-svg-icons": "^6.6.0",
    "@fortawesome/pro-solid-svg-icons": "^6.6.0",
    "@fortawesome/react-fontawesome": "^0.2.2",
    "@fortawesome/sharp-light-svg-icons": "^6.6.0",
   ```
   
   Reiterating a need for this fix, as it prevents standard updates.

**Comment by liz-gunther** [comment]
Also interested in a fix for this -- @CarrieWDing's solution worked for updating the package and the app compiles/runs fine, but we're having Cypress tests fail due to the unresolved internal dependency. I can't share the exact error message (I'm on my personal GH but the issue is happening in a project for work), but it's more or less as follows:

```
[webpack-dev-server] ERROR in ./x/x/x/
...
Error: /path/to/my/file.tsx: Cannot find module '@fortawesome-internal-tools/fontawesome-icons/canonical'
.... more of the stacktrace ....
```

Sounds like the issue is known so hopefully a fix will be in place soon :)

alternatively, if anyone can suggest a workaround for this in the meantime, I'm all ears!

**Comment by tomerh2001** [comment]
Any updates? Downgrading all packages doesn't work for me, I still get the same error...

**Comment by robmadole** [comment]
No update yet, it's still on our list to be fixed. I will mention that we've had enough trouble with these Babel macros that we are making the call to deprecate them in favor of something more reliable. We've already updated our docs accordingly.

@tomerh2001 if you are still getting the error it means that you haven't actually downgraded. Can you double-check?

**Comment by tomerh2001** [comment]
> No update yet, it's still on our list to be fixed. I will mention that we've had enough trouble with these Babel macros that we are making the call to deprecate them in favor of something more reliable. We've already updated our docs accordingly.
> 
> @tomerh2001 if you are still getting the error it means that you haven't actually downgraded. Can you double-check?

Thanks, I just missed the `^`. Adding it fixed it for now

**Comment by jeffgaynor** [comment]
This issue is still happening in v6.7.1. I'm stuck at v6.5.2 for now. When is a fix for this planned?

**Comment by MeshyIce** [comment]
I am encountering this issue for v6.7.2 as well. It also happened to me at v6.5.1 and v6.5.2.
Has anyone found a fix around this issue?

**Comment by jowyongrsui** [comment]
For anyone still encountering this issue, we got around this (and bumped our packages to version ^6.7.2) by including the specific libraries as dependencies and referencing them directly without using the import macro eg.

Instead of:
```
import { solid } from "@fortawesome/fontawesome-svg-core/import.macro";
...
<FontAwesomeIcon icon={solid("exclamation-circle")} />
```

We updated our code to:
```
import { faExclamationCircle } from "@fortawesome/pro-solid-svg-icons";
...
<FontAwesomeIcon icon={faExclamationCircle} />
```

**Comment by rayraspberry** [comment]
this still seems to be an issue during a standard `yarn` on a fresh system

`Cannot find module '@fortawesome-internal-tools/fontawesome-icons/canonical'`

Changing `package.json` lines to:
```
   "@fortawesome/fontawesome-svg-core": "6.5.2",
    "@fortawesome/free-brands-svg-icons": "6.5.2",
    "@fortawesome/free-solid-svg-icons": "6.5.2",
```

Seems to allow the site to build

**Comment by amohar** [comment]
Yeah, it seems it's still active. As I'm using Font-Awesome Pro, changing to v6.5.2 didn't seem to have helped. If it helps anyone, I have a fix, but it is really an ugly one:

1. I shimmed all the macros like so:
```
import { findIconDefinition, IconDefinition, IconName, parse } from "@fortawesome/fontawesome-svg-core";

export function brands(iconName: IconName): IconDefinition {
    return findIconDefinition(parse.icon(`fa-brands fa-${iconName}`));
}

export function duotone(iconName: IconName): IconDefinition {
    return findIconDefinition(parse.icon(`fa-duotone fa-${iconName}`));
}

export function light(iconName: IconName): IconDefinition {
    return findIconDefinition(parse.icon(`fa-light fa-${iconName}`));
}

export function regular(iconName: IconName): IconDefinition {
    return findIconDefinition(parse.icon(`fa-regular fa-${iconName}`));
}

export function solid(iconName: IconName): IconDefinition {
    return findIconDefinition(parse.icon(`fa-solid fa-${iconName}`));
}

export function thin(iconName: IconName): IconDefinition {
    return findIconDefinition(parse.icon(`fa-thin fa-${iconName}`));
}
```
2. I needed to go everywhere where I'm referencing a macro and switch the reference to my shims, and this is not even the ugliest part
3. The ugliest part - I needed to include the whole icon libraries so that I can freely use the shims:
```
import { fas } from '@fortawesome/pro-solid-svg-icons';
import { far } from '@fortawesome/pro-regular-svg-icons';
import { fal } from '@fortawesome/pro-light-svg-icons';
import { IconPack, library } from '@fortawesome/fontawesome-svg-core'

...

library.add(fas as IconPack);
library.add(fal as IconPack);
library.add(far as IconPack);
```

Very ugly. Could you guys please look into fixing this? Thanks!

**Comment by allsyntax** [comment]
Will upgrading to 7.x resolve the issue?

**Comment by kellow77** [comment]
This issue is still outstanding. Upgrading to 7.x does not resolve it. Is there any updates on this?

**Comment by robmadole** [comment]
Hey folks. I'm making the call that we won't be fixing this. This macro has been nothing but a support nightmare and frustrating to most folks who've I've chatted with who use it.

So this bug persists and we'll be removing the macro entirely in a minor release of 7.x.

For those who are still trying to use this, can you move to another method documented of importing icons?

**Comment by allsyntax** [comment]
I thought the issue was a coding reference to an internal diagnostic tool? I understand not supporting the method of importing icons in the future, but Ya'll should at least patch the bug.

**Comment by fulutas** [comment]
> > Hey guys, is there an update on this or a workaround? We have 6.5.2 installed and are having this issue.
> 
> We resolved the issue by removing the `^` symbol before the version number in the `package.json`, and then reinstalling the `node_modules`. Hope it helps.

That worked; thank you for your help.
